go/vt/vttls/vttls.go

```diff
@@ -94,17 +94,23 @@ func ClientConfig(cert, key, ca, name string) (*tls.Config, error) {
 
 // ServerConfig returns the TLS config to use for a server to
 // accept client connections.
-func ServerConfig(cert, key, ca string) (*tls.Config, error) {
+func ServerConfig(cert, key, ca, serverCA string) (*tls.Config, error) {
 	config := newTLSConfig()
 
-	certificates, err := loadTLSCertificate(cert, key)
-
-	if err != nil {
-		return nil, err
+	if serverCA != "" {
+	    certificates, err := combineAndLoadTLSCertificates(serverCA, cert, key)
+	    if err != nil {
+		    return nil, err
+	    }
+	    config.Certificates = *certificates
+	} else {
+	    certificates, err := loadTLSCertificate(cert, key)
+	    if err != nil {
+		    return nil, err
+	    }
+	    config.Certificates = *certificates
```

Comment:
If you wanna avoid some duplication, this can be shortened to

```
var certificates []*tls.Certificate
var err error

if serverCA != "" {
	    certificates, err := combineAndLoadTLSCertificates(serverCA, cert, key)
	} else {
	    certificates, err := loadTLSCertificate(cert, key)
}

if err != nil {
		    return nil, err
	    }
	    config.Certificates = *certificates
	    ```

Reply:
thanks, fixed